homeassistant/components/vacuum/neato.py

```diff
@@ -101,6 +101,10 @@ def update(self):
                 self.robot.state['action'] == 3 and
                 self.robot.state['state'] == 2):
             self._clean_state = STATE_ON
+        elif (self.robot.state['action'] == 11 or
+              self.robot.state['action'] == 12 and
+              self.robot.state['state'] == 2):
```

Comment:
Each of these `robot.state` accesses will make a request. That seems very inefficient. We should cache the json retrieved from the first access and use that in the following checks.

Reply:
Thank you so much for your comments @MartinHjelmare I was actually wondering why there was a burst of calls being made each time and this definitely tells me why :) I will go ahead and test these changes and check the rest of the components to make sure that we are using the cached state in the `switch` and `camera` as well.  Will submit another PR once ready.  Thanks again for pointing this out!

Reply:
Super!